mitosheet/src/mito/Mito.tsx

```diff
@@ -279,15 +275,64 @@ export const Mito = (props: MitoProps): JSX.Element => {
         }
     }, [mitoAPI, sendFunctionStatus])
 
+    // We're storing the last analysisData in a ref so that we can check the
+    // analysisData's code against the code in the cell and make sure we aren't
+    // overwriting any changes the user might have made. 
+    const oldCodeRef = useRef<string[] | undefined>();
     useEffect(() => {
-        /**
-         * We only write code after the render count has been incremented once, which
-         * means that we have read in and replayed the updated analysis, etc. 
-         */
-        if (analysisData.renderCount >= 1) {
-            // Finially, we can go and write the code!
-            props.jupyterUtils?.writeGeneratedCodeToCell(analysisData.analysisName, analysisData.code, userProfile.telemetryEnabled, analysisData.publicInterfaceVersion);
+        // If the oldCodeRef.current is undefined, then we haven't 
+        // loaded the code from the saved_analysis yet. We want to load
+        // this first in case Mito has been updated and the code generated
+        // by Mito has changed. If we don't do this, it's hard to differentiate
+        // between user changes and changes made by Mito to the generated code.
+        // The next time we try to write code, it will use this saved analysis code
+        // to check against. 
+        if (oldCodeRef.current === undefined) {
+            void mitoAPI.getSavedAnalysisCode().then((response: MitoAPIResult<string[]>) => {
+                if ('error' in response) {
+                    console.error(response.error);
+                    return;
+                }
+                oldCodeRef.current = response.result;
+            });
+            // After setting the oldCodeRef, we return because this is supposed to represent
+            // the code from the last time analysisData changed
+            return;
         }
+    
+        void props.jupyterUtils?.writeGeneratedCodeToCell(
```

Comment:
Do we need this function to be async anymore?

Reply:
nope